lib/kubernetes-deploy/resource_cache.rb

```diff
@@ -36,6 +36,11 @@ def get_all(kind, selector = nil)
       []
     end
 
+    def prewarm(resources)
+      kinds = (resources.map(&:type) + resources.flat_map(&:prefetch_kinds_on_sync)).uniq
+      KubernetesDeploy::Concurrency.split_across_threads(kinds) { |kind| use_or_populate_cache(kind) }
```

Comment:
I think we want to use the public `get_all` instead. If we don't, we need to handle the `KubectlError`.

Reply:
Also, since we know this is all IO, we should probably crank up the concurrency instead of using even splitting like this. You could make `MAX_THREAD` configurable instead of constant, and set it to the number of kinds we have?